api/types/okta_test.go

```diff
@@ -79,6 +80,187 @@ func TestOktaAssignments_SetStatus(t *testing.T) {
 	}
 }
 
+func TestOktAssignmentIsEqual(t *testing.T) {
```

Comment:
I feel like we may need to take a fuzzier approach to this where the Test uses reflection to try modifying all fields one by one bar those which are excluded - in addition, this would probably reduce the overhead involved in doing this for new resources. It feels like it's far too easy to miss a new field off in the comparison and the reconciler behaviour could get super confusing.

Reply:
Yup - that's almost exactly what I'd suggest. We can probably make a fairly nice test helper that'll just fuzz all fields bar those excluded and evaluate some arbitrary predicate for each one.

Reply:
So I've been toying around with this. I don't know that this is quite as easy as the example there -- our resources can be nested, arbitrary values, etc. Generalizing this would be fairly difficult and I'm worried it would be brittle.

Now, if we used goderive, we could potentially introduce a CI job to test these things. Thoughts?